python/mxnet/module/base_module.py

```diff
@@ -458,11 +458,22 @@ def fit(self, train_data, eval_data=None, eval_metric='acc',
         for epoch in range(begin_epoch, num_epoch):
             tic = time.time()
             eval_metric.reset()
-            for nbatch, data_batch in enumerate(train_data):
+            nbatch = 0
+            end_of_batch = False
+            next_data_batch = train_data.next()
```

Comment:
use iter. don't call next directly. Someone might override iter_next
http://stackoverflow.com/questions/16814984/python-list-iterator-behavior-and-nextiterator